packages/lab/src/calendar/internal/CalendarDay.tsx

```diff
@@ -32,91 +32,92 @@ export interface CalendarDayProps<TDate>
 
 const withBaseName = makePrefixer("saltCalendarDay");
 
-export const CalendarDay = forwardRef<HTMLButtonElement, CalendarDayProps<any>>(
-  function CalendarDay<TDate extends DateFrameworkType>(
-    props: CalendarDayProps<TDate>,
-    ref: React.Ref<HTMLButtonElement>,
-  ) {
-    const {
-      className,
-      day,
-      renderDayContents,
-      month,
-      TooltipProps,
-      format = "DD",
-      ...rest
-    } = props;
-    const { dateAdapter } = useLocalization<TDate>();
-    const targetWindow = useWindow();
-    useComponentCssInjection({
-      testId: "salt-calendar-day",
-      css: calendarDayCss,
-      window: targetWindow,
-    });
+export const CalendarDay = forwardRef<
+  HTMLButtonElement,
+  CalendarDayProps<DateFrameworkType>
+>(function CalendarDay<TDate extends DateFrameworkType>(
+  props: CalendarDayProps<TDate>,
+  ref: React.Ref<HTMLButtonElement>,
+) {
+  const {
+    className,
+    day,
+    renderDayContents,
```

Comment:
Maybe this is what unselectable range day SO question's answer

Reply:
Refactored this API, with the latest  changes to `renderDayButton` it is. Check the examples in the docs, if interested in this feature.